common/src/main/java/com/wynntils/models/spells/SpellCasterModel.java

```diff
@@ -0,0 +1,50 @@
+package com.wynntils.models.spells;
+
+import com.wynntils.core.components.Model;
+import com.wynntils.mc.event.ChangeCarriedItemEvent;
+import com.wynntils.models.spells.type.SpellDirection;
+import com.wynntils.models.worlds.event.WorldStateEvent;
+import net.neoforged.bus.api.SubscribeEvent;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
+
+public final class SpellCasterModel extends Model {
```

Comment:
Now that this is extracted into a separate class, it seems to be much closer to a Handler functionality than a model. What do you think about moving this into `SpellCastHandler`?

Reply:
I did consider this, but hesitated as I didn't feel like it matched too closely to the description in the abstract Handler description. Good that you also think it can be a handler, I'll change it